src/Resources/app/storefront/src/checkout/confirm-order.plugin.js

```diff
@@ -11,13 +11,53 @@ export default class ConfirmOrderPlugin extends Plugin {
 
     init() {
         this._client = new StoreApiClient();
+        const { locale, originKey, clientKey, environment, paymentMethodsResponse } = adyenCheckoutConfiguration;
+        const ADYEN_CHECKOUT_CONFIG = {
+            locale,
+            originKey,
+            clientKey,
+            environment,
+            showPayButton: false,
+            hasHolderName: true,
+            paymentMethodsResponse: JSON.parse(paymentMethodsResponse),
+            onAdditionalDetails: this.handleOnAdditionalDetails.bind(this)
+        };
+
+        this.adyenCheckout = new AdyenCheckout(ADYEN_CHECKOUT_CONFIG);
         this.confirmOrderForm = DomAccess.querySelector(document,
             '#confirmOrderForm');
         this.confirmOrderForm.addEventListener('submit',
             this.validateAndConfirmOrder.bind(this));
         this.initializeCustomPayButton();
     }
 
+    handleOnAdditionalDetails (state) {
```

Comment:
Will it be possible to keep only one handleOnAdditionalDetails function for confirm-order.plugin and checkout.plugin?

Reply:
Yes, since we have a ticket (PW-4108) for this refactoring, I thought it's best to only make the changes I needed to make for now